CASSANDRA-19954 add allowAutoSnapshots nor incrementalBackups in TableParams.Serializer.serialize / deserialize

smiklosovic opened the pull request:
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

dcapwell commented on src/java/org/apache/cassandra/tcm/serialization/Version.java:
```diff
@@ -38,6 +38,10 @@ public enum Version
+    /**
+     *  - Serialize allowAutoSnapshot and incrementalBackups when serializing TableParams
+     */
+    V3(3),
```
think you need to update `org.apache.cassandra.tcm.membership.NodeVersion#CURRENT_METADATA_VERSION`?

dcapwell reviewed (commented):
mostly LGTM, think `NodeVersion.CURRENT_METADATA_VERSION` needs to point to `v3`...

ill let the TCM folks review and make sure everything makes sense

krummas reviewed (changes requested):
lgtm in general, but yep, need to bump `CURRENT_METADATA_VERSION`

and perhaps add a test showing this regression is fixed?

smiklosovic committed:
Make TableParams.Serializer set allowAutoSnapshots and incrementalBackups

patch by Stefan Miklosovic; reviwed by Marcus Eriksson, David Capwell for CASSANDRA-19954
